magda-sleuther-format-layerer/src/onRecordFound.ts

```diff
@@ -0,0 +1,113 @@
+import getExtensionProcessedData from "./format-engine/measures/processed-functions/extensionProcessedFns";
+
+import  getBestMeasureResult  from "./format-engine/measureEvaluatorByHierarchy"
+import MeasureEvaluationSet from "./format-engine/measures/MeasureEvaluationSet";
+import MeasureEvalResult from "./format-engine/MeasureEvalResult";
+
+//import { Snapshot } from "../../magda-typescript-common/src/format/MeasureSnapShot";
+let mochaObject = {
+    isRunning: true // set to false if not testing this function onRecordFound
+}
+export default async function onRecordFound(
+    record: Record,
+    registry: Registry
+) {
+    const distributions: Record[] =
+        record.aspects["dataset-distributions"] &&
+        record.aspects["dataset-distributions"].distributions;
+
+    if (!distributions || distributions.length === 0) {
+        return Promise.resolve();
+    }
+
+    //TODO delete this piece of code
+    console.log("got after distribution nul check");
+
+    // 2D array: 1 row per distribution
+    const retrievedEvalSets: MeasureEvaluationSet[][] = distributions.map(function (distribution) {
+        const dcatSet: MeasureEvaluationSet = {
+            measureResult: getDcatMeasureResult(distribution),
+            getProcessedData: getDcatProcessedData
+        }
+        const extensionSet: MeasureEvaluationSet = {
+            measureResult: getExtensionMeasureResult(distribution),
+            getProcessedData: getExtensionProcessedData
+        }
+        const downloadSet: MeasureEvaluationSet = {
+            measureResult: getDownloadMeasureResult(distribution),
+            getProcessedData: getDownloadProcessedData
+        }
+
+        //TODO delete this
+        console.log("the list recieved is: " + [dcatSet, extensionSet, downloadSet].toString() + "for the distribution: " + distribution.id)
+
+
+        return [dcatSet, extensionSet, downloadSet];
+
+    });
+
+    const bestFormatResults: MeasureEvalResult[] = retrievedEvalSets.map(evalSetsPerDist =>
+        getBestMeasureResult(evalSetsPerDist)
+    );
+
+    //TODO delete
+    console.log("best results obtained" + bestFormatResults.toString());
+
+    bestFormatResults.forEach(function(formatResult) {
+        !mochaObject.isRunning ?
+        recordFormatAspect(
+            registry,
+            formatResult.distribution,
+            {
+                format: formatResult.format.format,
+                confidenceLevel: formatResult.absConfidenceLevel
+            }
+        ) : recordFormatAspectTest(
+            registry,
+            formatResult.distribution,
+            {
+                format: formatResult.format.format,
+                confidenceLevel: formatResult.absConfidenceLevel
+            }
+        )
+    });
+
+    //TODO delete this piece of code
+    console.log("the best results gathered were: " + bestFormatResults.toString());
+
+    Promise.resolve();
```

Comment:
Is there supposed to be a `return` here? In any case doesn't look like you need it.